src/main/java/ar/com/carloscurotto/storm/complex/topology/propagator/gloss/MessageSender.java

```diff
@@ -0,0 +1,118 @@
+package ar.com.carloscurotto.storm.complex.topology.propagator.gloss;
+
+import java.io.StringWriter;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import javax.xml.bind.JAXBContext;
+import javax.xml.bind.JAXBException;
+import javax.xml.bind.Marshaller;
+import javax.xml.transform.stream.StreamResult;
+
+import org.apache.commons.lang3.Validate;
+
+import ar.com.carloscurotto.storm.complex.service.OpenAwareService;
+import ar.com.carloscurotto.storm.complex.topology.propagator.gloss.message.ExcpTradeStatusMessage;
+import ar.com.carloscurotto.storm.complex.topology.propagator.gloss.message.NormalTradeStatusMessage;
+import ar.com.carloscurotto.storm.complex.topology.propagator.gloss.message.TradeCommentsMessage;
+import ar.com.carloscurotto.storm.complex.topology.propagator.gloss.message.TradeMessage;
+import ar.com.carloscurotto.storm.complex.transport.Producer;
+
+/**
+ * Sends the message into the queue. Knows what queue to use and knows how to marshall the messages.
+ *
+ * @author D540601
+ */
+public class MessageSender extends OpenAwareService<List<TradeMessage>, Void> {
+
+    protected Map<Class<? extends TradeMessage>, Marshaller> marshallers = new HashMap<Class<? extends TradeMessage>, Marshaller>();
+
+    private Producer<String> messageProducer;
+
+    /**
+     * Constructs the MessageSender with the give message producer.
+     * 
+     * @param theMessageProducer
+     *            a {@link Producer<String>} that will be used to send the update messages to the consumer of the
+     *            producer.
+     */
+    public MessageSender(final Producer<String> theMessageProducer) {
+        Validate.notNull(theMessageProducer, "The message producer cannot be null");
+        messageProducer = theMessageProducer;
+    }
+
+    /**
+     * Creates the marshallers for the trade status messages classes and stores them in the marshallers map.
+     */
+    @Override
+    protected void doOpen() {
+        try {
+            marshallers.put(NormalTradeStatusMessage.class,
```

Comment:
Ahora que usamos IoC, esta clase podria recibir en el constructor una lista de clases que vas a usar como marshallers.
Luego en el doOpen solo harias:

for(Class<?>marshaller : marshallers) JAXBContext.newInstance(marshaller).createMarshaller();